### PR TITLE
Implement data ocean model for moab driver 

### PR DESCRIPTION
**Pull request by iulian787**
Add support for  the MOAB coupler in data-ocean.

This allows running F-cases that have ocean as the only data model and use MPAS-Seaice with the moab coupler.

On the component data side, instance a new imoab application whose degrees of freedom are aligned with the regular 
 domain  decomposition from the data model.  More changes were needed on the coupler side, because previous coupled cases did not have samegrid_ao flag true. 
 Also, changes in the intel compiler or bios on chrysalis uncovered other needs about initialization of
mpi_reduce variables. (mpi_reduce is needed for comparisons between mct and moab fields,
 in debug/verification mode)

Also add MOAB libs to perlmutter.

[BFB]

### CODE REVIEW

File: cime_config/machines/cmake_macros/intel_pm-cpu.cmake
Comment (rljacob) [inline cime_config/machines/cmake_macros/intel_pm-cpu.cmake:32]: What happens if you don't set MOAB_PATH?  Jim has said we shouldn't need both MOAB_PATH and MOAB_ROOT.

Comment (rljacob) [reply cime_config/machines/cmake_macros/intel_pm-cpu.cmake:32]: update:  will remove these later.

File: cime_config/machines/cmake_macros/gnu_anlgce.cmake
Comment (rljacob) [reply cime_config/machines/cmake_macros/gnu_anlgce.cmake:3]: I think I have to wait until you merge #6308 because its needs the changes to the MPAS framework cmake.  I'll rebase this after you merge 6308.

**Comment by rljacob** [comment]
@jgfouca the MOAB_PATH's were removed and this was rebased on to the head of master.  F-cases still build and run on perlmutter with moab.
tested with:  ./create_newcase --case F1850mb --res ne4pg2_oQU480 --compset F1850 --driver moab

File: cime_config/machines/cmake_macros/gnu_anlgce.cmake
Comment (rljacob) [reply cime_config/machines/cmake_macros/gnu_anlgce.cmake]: I've rebased this branch and removed all the MOAB_PATHs

**Review by jgfouca — APPROVED** [review]

**Pull request merged**
